presto-orc/src/main/java/com/facebook/presto/orc/reader/ListStreamReader.java

```diff
@@ -73,7 +75,7 @@
     // The set of subscripts for which data needs to be
     // returned. Other positions can be initialized to null. If this
     // is null, values for all subscripts must be returned.
-    long[] subscripts;
+    private int[] subscripts;
```

Comment:
Is it because we can never have such a ridiculously high subscript?  If so, why are we using LongSubscript instead of for example IntegerSubscript?

Reply:
@tdcmeehan LongSubscript is used for both array subscript (where values can only be integers) and map subscript (where values can be both integers and longs, e.g. map<integer, T> and map<bigint, T>).

Reply:
Do you think it makes sense to differentiate it when we parse?  From my reading we should be able to tell the difference when we construct the `XxSubscript` (correct me if I'm wrong?)

Reply:
@tdcmeehan At this point, I don't see any practical value of doing that. It would require examining the type of the SubscriptExpression#base, which is not readily available.